src/node/config.h

```diff
@@ -15,8 +15,13 @@ namespace ccf
     size_t recovery_threshold = 0;
 
     ConsensusType consensus = ConsensusType::CFT;
+
     std::optional<ReconfigurationType> reconfiguration_type = std::nullopt;
 
+    // If true, the service endorses the certificate of new trusted nodes, and
```

Comment:
I'm confused as to why this is optional and not just boolean.

Reply:
I started a discussion on this just now: https://github.com/microsoft/CCF/discussions/2916 I agree that having the field as non-optional is better but I think there are implications for ledger serialisation.